src/posthog-core.ts

```diff
@@ -386,15 +386,31 @@ export class PostHog {
         this._gdpr_init()
 
         if (config.segment) {
-            // Use segments anonymousId instead
-            this.config.get_device_id = () => config.segment.user().anonymousId()
-
-            // If a segment user ID exists, set it as the distinct_id
-            if (config.segment.user().id()) {
-                this.register({
-                    distinct_id: config.segment.user().id(),
+            // If segment analytics.js 2.0
```

Comment:
I don't think this is correct. We tested this and there's a reason why we document the integration specifically [this way](https://posthog.com/docs/libraries/segment#enabling-all-features-via-segment-with-your-website) to ensure that `.ready()` has been used. Once that is used, the `user()` function returns the User rather than a promise. (They mention it explicitly here https://github.com/segmentio/analytics-next/blob/master/packages/browser/src/core/analytics/interfaces.ts#L89) 

Have you tried implementing this following the instructions accurately and are _still_ seeing it as a promise?